src/app/common/schemas.py

```diff
@@ -125,28 +185,112 @@ def load_from_fitz_rect(rect: fitz.Rect) -> "BoundingBox":
+        ...,
+        description="""
+            Northing coordinate, representing the vertical position of the point. The value should be in the 
+            units of the specified projection system.
+        """,
+        example=2.0,
+    )
+    projection: str = Field(
+        ...,
+        description="""
+            Projection system used to reference the coordinates. This defines the coordinate reference system, 
+            such as 'LV95' for Swiss coordinate systems.
+        """,
+        example="LV95",
+    )
 
 
 class ExtractDataRequest(ABC, BaseModel):
-    """Request schema for the extract_data endpoint.
+    """Request schema for the `extract_data` endpoint.
+
+    **Coordinate Systems:**
+    - **PNG coordinates:** Pixels are measured from the top-left corner (0, 0), where x increases rightward
+    and y downward.
+    - **PDF coordinates:** Also measured from the top-left corner (0, 0), though any transformations between
+    PDF and PNG coordinates are managed internally by the `BoundingBox.rescale` method.
+
+    ### Fields
+    Each field below includes inline examples to aid users in creating requests. See `json_schema_extra`
+    for a complete example.
+
+    **Attributes:**
+    - **filename** (`Path`): Path to the file (PNG or PDF). _Example_: `"document.png"`
+    - **page_number** (`int`): Target page for data extraction. This is a 1-based index. _Example_: `1`
+    - **bbox** (`BoundingBox`): Bounding box for the extraction area, in PNG coordinates. Origin is the
+    top-left, with x increasing rightward and y increasing downward.
+        - Example format: `{"x0": 0.0, "y0": 0.0, "x1": 100.0, "y1": 100.0}`
+    - **format** (`FormatTypes`): Specifies the expected format for extracted data, e.g., `"coordinates"`.
+
+    ### Validation
+    Custom validators ensure data integrity:
+    - **Filename Validator:** Ensures filename is not empty.
+    - **Page Number Validator:** Confirms page number is positive.
+    - **Format Validator:** Checks format is valid as per `FormatTypes`.
+
+
+    The bounding box should be provided in PNG coordinates. Any necessary coordinate transformations between PNG
+    and PDF are handled internally using the BoundingBox.rescale method.
 
     Each field in the Pydantic model can have an example parameter, which provides an inline
     example for that specific field.
     """
 
-    filename: Path = Field(..., example=Path("document.png"))
-    page_number: int = Field(..., example=1)  # 1-based index
-    bbox: BoundingBox = Field(..., example={"x0": 0.0, "y0": 0.0, "x1": 100.0, "y1": 100.0})
-    format: FormatTypes = Field(..., example=FormatTypes.COORDINATES.value)
+    filename: Path = Field(
+        ...,
+        description="""
+            Path to the input document file (PNG or PDF) that contains the data to be extracted. This should be
+            a valid file path, and the file should be accessible to the API.
+        """,
+        example=Path("document.png"),
```

Comment:
Again, only PDF supported, as far as I understand the code?

Additionally, the PDF file must have been processed by the create_pngs endpoint first, right? We should mention that in the docs as well.

Reply:
Fully agree, this was not clear enough. I made a suggestion in the next commit. I added it to the schema here as well as in the `router.py` in the description of the `post_extract_data` endpoint.